setup/ProjectSystemSetup/ProjectSystemSetup.csproj

```diff
@@ -55,11 +55,16 @@
     </Compile>
   </ItemGroup>
 
-  <!-- Excludes the localized .xaml files from the VSIX which are added from the GetCopyToOutputDirectoryItems output group. Only the localized .dlls are needed in the VSIX. -->
-  <!-- For additional info, see: https://github.com/dotnet/project-system/issues/6612 -->
-  <Target Name="ExcludeLocalizedXamlFilesFromVSIX" AfterTargets="GetVsixSourceItems">
+  <Target Name="PostProcessVsixSourceItems" AfterTargets="GetVsixSourceItems">
     <ItemGroup>
+      <!-- Excludes the localized .xaml files from the VSIX which are added from the GetCopyToOutputDirectoryItems output group. Only the localized .dlls are needed in the VSIX. -->
+      <!-- For additional info, see: https://github.com/dotnet/project-system/issues/6612 -->
       <VSIXSourceItem Remove="$(ArtifactsObjDir)$(_ManagedProjectName)\$(TargetFramework)\$(_ManagedProjectName).xlf\**\*.xaml" />
+
+      <VSIXSourceItem Update="@(VSIXSourceItem)"
```

Comment:
Per https://github.com/dotnet/msbuild/issues/2835, using `Update` in a target does not do what you expect.

Reply:
It seems to work. 🤷

Reply:
Updates in targets unconditionally set metadata on ALL items of that type. So it will disable NGen for all items in the VSIX.